File: func1.py
# func1

# // const nextCharForNumberString = str => {
# //     const trimmed = str.trim();
# //     const number = parseInt(trimmed);
# //     const nextNumber = number + 1;
# //     return String.fromCharCode(nextNumber)
# // }

class Box:
    def __init__(self, x):
        self.x = x
        self.map = lambda f: Box(f(x))
        self.fold = lambda f: f(x)
    
    def __repr__(self):
        return 'Box({})'.format(self.x)

# Box = lambda x: {
#     'map': lambda f: Box(f(x))
#     'fold': lambda f: f(x)
# }

# def nextCharForNumberString(st):
#     return map(lambda i: chr(i), map(lambda n: n+1, map(lambda x: int(x), map(lambda s: s.strip(), [st]))))

def nextCharForNumberString(st):
    return Box(st).map(lambda s: s.strip()) \
        .map(lambda x: int(x)) \
        .map(lambda n: n+1) \
        .fold(lambda i: chr(i)) \


result = nextCharForNumberString(' 64 ')

print(result)

File: func1.js
// Func JS 1: Box

// const nextCharForNumberString = str => {
//     const trimmed = str.trim();
//     const number = parseInt(trimmed);
//     const nextNumber = number + 1;
//     return String.fromCharCode(nextNumber)
// }

const Box = x => ({
    map: f => Box(f(x)),
    fold: f => f(x),
    inspect: () => `Box(${x})`
})

const nextCharForNumberString = str =>
    Box(str)
    .map(s => s.trim())
    .map(r => parseInt(r))
    .map(i => i + 1)
    .fold(j => String.fromCharCode(j))



const result = nextCharForNumberString(' 64 ');

console.log(result)